tests/integration/test_bundle.py

```diff
@@ -196,3 +199,33 @@ async def test_bundle_charms_can_handle_frequent_update_status(ops_test: OpsTest
 
     # After update-status frequency is restored to default, make sure all charms settle into active
     await ops_test.model.wait_for_idle(status="active")
+
+
+async def test_prometheus_scrapes_loki_through_traefik(ops_test: OpsTest):
+    """Prometheus should correctly scrape Loki through its traefik endpoint."""
+    prom_url = await get_proxied_unit_url(ops_test, "prometheus", 0)
+
+    response = urllib.request.urlopen(f"{prom_url}/api/v1/targets", data=None, timeout=2.0)
+    assert response.code == 200
+    targets = json.loads(response.read())["data"]["activeTargets"]
+    targets_summary = [(t["discoveredLabels"]["__metrics_path__"], t["health"]) for t in targets]
+    assert (f"/{ops_test.model.name}-loki-0/metrics", "up") in targets_summary
+    logger.info("prometheus is successfully scraping loki through traefik")
```

Comment:
Could this be [simplified](https://github.com/canonical/prometheus-k8s-operator/blob/1a25f576ed3afb5965a44cdaaa2c768d0dcfa00a/tests/integration/test_external_url.py#L190) by using a regular string search?
```
response = response.read()
assert '"health":"up"' in response
assert '"health":"down"' not in response
```

Reply:
Not really, as the response contains the status for all the targets; checking that everything is up would misrepresent the test which is only checking for loki